service/src/main/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsService.java

```diff
@@ -89,11 +92,11 @@ private String getCourtName(String courtName) {
         HearingVenue hearingVenue = hearingVenueLookUpService.getHearingVenue(prioritisedHearingBooking.getVenue());
 
         return ImmutableMap.of(
-            "hearingDate", dateFormatterService.formatLocalDateToString(prioritisedHearingBooking.getDate(),
-                FormatStyle.LONG),
+            "hearingDate", commonCaseDataExtractionService.getHearingDate(prioritisedHearingBooking),
```

Comment:
Personally I am not fan of this implementation - it's not obvious what's the hidden logic and in the template that if the hearingDate is the same we don't return anything in this field and we will hide one row. 
Wouldn't adding the some boolean flag make it more readable? What are your thoughts?

Reply:
Maybe `getHearingDate` is the wrong name, maybe `getFormattedHearingDate`. Could extract the same date check that is done internally in the method to a parameter.

Reply:
Changed name to be more clear and now returns optional